### ISSUE TITLE
Highlight code sections

### ISSUE THREAD

**Issue by WolfX2** [dbeaver/dbeaver#13239]
**Is your feature request related to a problem? Please describe.**
There may be multiple places i repeat a piece of code in one query - for example i could write:
`TO_NUMBER(TO_CHAR(SYSDATE-1, 'YYYYMMDD'))`
In multiple places, but if i want to change the date range, i would likely have to change it in all places in occours.

**Describe the solution you'd like**
It would be nice if you could have a piece of code remain consistently highlighted (with a background color of my choice). For example, in this case, i could 'pin' the `TO_NUMBER(TO_CHAR(SYSDATE` portion of the script, so it would appear highlighted everywhere in the SQL file, just so this is easy to find throughout the file.

Something like this:

![image](https://user-images.githubusercontent.com/30268352/126014123-e7534dda-98e7-4209-8b1a-8dcb15dff88f.png)

**Describe alternatives you've considered**
NA

**Additional context**
NA

**Comment by Matvey16** [comment]
You can already find and even replace everything at once with Find/Replace feature using Ctrl+F
Are you suggesting this feature because it will be more convenient?

**Comment by WolfX2** [comment]
> You can already find and even replace everything at once with Find/Replace feature using Ctrl+F
> Are you suggesting this feature because it will be more convenient?

Yes, i am specifically looking for functionality to highlight or call out important or significant, blocks or lines of code. This is different functionality than the find/replace function.

**Comment by Matvey16** [comment]
@WolfX2 So just to clarify, you suggest a feature that will highlight all instances of a specific part of code. Just highlight with a chosen background color, without any changes to text. And this is to help navigating in a long script. Correct?

**Comment by WolfX2** [comment]
> @WolfX2 So just to clarify, you suggest a feature that will highlight all instances of a specific part of code. Just highlight with a chosen background color, without any changes to text. And this is to help navigating in a long script. Correct?

Thats correct. You could compare this to the well-known functionality that exists in notepad++ where you can style all, or one occurrence of a code block in the context menu, screenshot attached.

![image](https://user-images.githubusercontent.com/30268352/130705137-e6282448-fab4-4e6a-b436-5206fff01742.png)

**Comment by Matvey16** [comment]
Thank you for a suggestion
